compositor_pipeline/src/pipeline/decoder/audio.rs

```diff
@@ -82,46 +66,107 @@ pub fn start_audio_decoder_thread(
     match decoder_init_result_receiver.recv() {
         Ok(Ok(())) => Ok(()),
         Ok(Err(init_err)) => Err(init_err),
-        Err(_recv_err) => Err(DecoderInitError::CannotReadInitResult),
+        Err(_recv_err) => Err(InputInitError::CannotReadInitResult),
     }
 }
 
-/// init_result_sender sends:
-/// - true init result for Opus
-/// - always ok for AAC (aac sample rate is unknown at register time, first chunk is need to determine it)
-fn run_decoder_thread(
-    opts: AudioDecoderOptions,
+fn run_resampler_only_thread(
+    input_sample_rate: u32,
     output_sample_rate: u32,
-    chunks_receiver: Receiver<PipelineEvent<EncodedChunk>>,
+    raw_samples_receiver: Receiver<PipelineEvent<DecodedSamples>>,
     samples_sender: Sender<PipelineEvent<InputSamples>>,
-    init_result_sender: Sender<Result<(), DecoderInitError>>,
+    init_result_sender: Sender<Result<(), InputInitError>>,
 ) {
-    // This ensures that EOS is send only once
-    let sender = |samples: InputSamples| {
-        if samples_sender.send(PipelineEvent::Data(samples)).is_err() {
-            debug!("Failed to send decoded input samples.");
-        };
+    let mut resampler = match Resampler::new(input_sample_rate, output_sample_rate) {
+        Ok(resampler) => {
+            if init_result_sender.send(Ok(())).is_err() {
+                error!("Failed to send rescaler init result.");
+            }
+            resampler
+        }
+        Err(err) => {
+            if init_result_sender.send(Err(err)).is_err() {
+                error!("Failed to send rescaler init result.");
+            }
+            return;
+        }
     };
 
-    run_decoding(
-        opts,
-        output_sample_rate,
-        chunks_receiver,
-        sender,
-        init_result_sender,
-    );
+    for event in raw_samples_receiver {
+        let PipelineEvent::Data(raw_sample) = event else {
+            break;
+        };
+
+        for input_samples in resampler.resample(raw_sample) {
+            if samples_sender
+                .send(PipelineEvent::Data(input_samples))
+                .is_err()
+            {
+                debug!("Failed to send decoded input samples.");
+            };
+        }
+    }
 
     if samples_sender.send(PipelineEvent::EOS).is_err() {
         debug!("Failed to send EOS message.")
     }
 }
 
+pub fn start_audio_decoder_thread(
```

Comment:
diff is a bit misleading here I added resampler funcitions above, only change here was inlining run_decoder_thread iniside this function